tests/nsrt_learning/strips_learning/test_gen_to_spec_learner.py

```diff
@@ -551,10 +570,28 @@ def test_backchaining_strips_learner_order_dependence():
     # Be sure to reset the segment add effects before doing this.
     learner.reset_all_segment_add_effs()
     reverse_order_pnads = learner.learn()
+    action_space = Box(0, 1, (1, ))
+    dataset = [traj2, traj1]
+    train_tasks = [task2, task1]
+    options = {
+        moveto_param_option, pick_param_option, movetoshelf_param_option,
+        place_param_option
+    }
+    ground_atom_dataset = utils.create_ground_atom_dataset(dataset, preds)
+    reverse_order_nsrts, _, _ = learn_nsrts_from_data(dataset,
+                                                      train_tasks,
+                                                      preds,
+                                                      options,
+                                                      action_space,
+                                                      ground_atom_dataset,
+                                                      sampler_learner="random")
 
     # First, check that the two sets of PNADs have the same number of PNADs.
     # Uh oh, they don't
     assert len(natural_order_pnads) != len(reverse_order_pnads)
+    # Second, check that the two sets of NSRTs have the same number of NSRTs.
+    # They do! Because of dataset reordering and harmless operator pruning.
+    assert len(natural_order_nsrts) == len(reverse_order_nsrts)
```

Comment:
wait - I'm a little confused by this: why do the pnad's have different lengths, but the NSRT's work? Could you augment this comment to explain more about what exactly is going on here?